### PR TITLE
ci: add draft-binaries workflow

### LINKED ISSUE #3829: Test ockam draft-binaries before they are released
We should test ockam binaries after they are released as a draft release across all supported platform (macos-latest and ubuntu-18.04). This would require us creating a workflow that
- Gets a `tag name` as input (name of the tag to test as draft release
- Download the binaries for the supported platform
- Test the binaries against our Bats example in `implementations/rust/ockam/ockam_command/tests` directory or simply print out a `ockam --version`

### PR DESCRIPTION
**Pull request by miles170**
## Proposed Changes

Closes #3829.

Add a workflow that tests ockam release binaries from the specified git tag. 
Only the x86_64 binaries are tested since there is no native arm-based GitHub runner.

Check passed CI here: https://github.com/miles170/ockam/actions/runs/3433956317

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

### CODE REVIEW

File: .github/workflows/draft-binaries.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      with:
+        fetch-depth: 0
+        ref: ${{ github.event.inputs.git_tag }}
+
+    - name: Download Binary And Signature
+      uses: robinraju/release-downloader@d6de084c58345d09b017e22701dbcf26977cfd14
+      with:
+        repository: build-trust/ockam
+        tag: ${{ github.event.inputs.git_tag }}
+        fileName: ockam.${{ matrix.target }}*
```
Comment (metaclips) [inline .github/workflows/draft-binaries.yml]:
```bash
gh release download ${{ github.event.inputs.git_tag }} -p ockam.${{ matrix.target }}* -R build-trust/ockam
```
Should replace this action. More here https://cli.github.com/manual/gh_release_download

File: .github/workflows/draft-binaries.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    - name: Verify Signature
+      run: |
+        curl --proto '=https' --tlsv1.2 -sSfL -o ockam.pub https://raw.githubusercontent.com/build-trust/ockam/develop/tools/docker/cosign.pub
+        cosign verify-blob --key ockam.pub --signature ockam.${{ matrix.target }}.sig ockam.${{ matrix.target }}
+
+    - name: Check Ockam Version
+      run: |
+        chmod +x ./ockam.${{ matrix.target }}
+        ./ockam.${{ matrix.target }} --version
```
Comment (metaclips) [inline .github/workflows/draft-binaries.yml]: This is a plus 🙂 and not really needed. We can `grep` to see if the version spewed out contains `${{ github.event.inputs.git_tag }}`

**Review by metaclips — CHANGES_REQUESTED** [review]
Thanks for an amazing PR @miles170. I left few suggestions for a merge

**Comment by miles170** [comment]
Check passed CI here: https://github.com/miles170/ockam/actions/runs/3438259508

**Review by metaclips — APPROVED** [review]
Thanks @miles170

**Review by adrianbenavides — APPROVED** [review]

**Pull request merged**